### PR TITLE
[CMDCT-3172] Convert EntityProvider to Zustand

### PR DESCRIPTION
**Pull request by karla-vm**
### Description

This covers the last part of our ✨ `#mcrZustandification` ✨ by converting the `EntityProvider` to Zustand. Some of these changes include refactoring decisions that were made while building out MFP.

### Related ticket(s)

CMDCT-3172

---
### How to test

There should be no changes in terms of functionality, so to test this, you should go through a MCPAR and MLR reports and be able to submit without any problems w/ entities.

### Notes

N/A

---
### Pre-review checklist

- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [ ] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Review by gmrabian — APPROVED** [review]
looks good

**Review by jessabean — DISMISSED** [review]
Looks good to me!

File: services/ui-src/src/utils/autosave/autosave.ts [hunk 1]
```diff
@@ -114,17 +112,14 @@ export const autosaveFieldData = async ({
   if (fieldsToSave.length) {
     const reportKeys = { reportType, id, state };
     let dataToWrite = {};
-    if (
-      entityContext &&
-      entityContext.selectedEntity &&
-      entityContext.entityType
-    ) {
+    if (entityContext && entityContext.selectedEntity) {
```
Comment (bangbay-bluetiger) [inline services/ui-src/src/utils/autosave/autosave.ts:115]: Shorten to `entityContext?.selectedEntity` ?

Comment (karla-vm) [reply services/ui-src/src/utils/autosave/autosave.ts]: good call! ✅

**Review by jessabean — APPROVED** [review]

**Review by bangbay-bluetiger — APPROVED** [review]

**Pull request merged**